lessons/cryptography/learn-to-count/learn-to-count.md

```diff
@@ -0,0 +1,105 @@
+
+The binary number system uses 2 as a base, but the values can be calculated in the same way that decimal numbers are calculated. 
+
+### Binary to Decimal Conversion
+For example, if we want to breakdown the binary number 0b10110, we can re-write the number as follows to convert it to a decimal number:
+
+0b10110 = 1*(2^4) + 0*(2^3) + 1*(2^2) + 1*(2^1) + 0*(2^0) = 8 + 0 + 4 + 2 + 0 = 14
+
+Note that the **0b** at the beginning of the number indicates that it is a binary number. This is to help distinguish between decimal numbers and binary numbers. 
+
+### Decimal to Binary Conversion
+You can also perform the opposite calculate to convert a decimal number to a binary number. In the image below, the decimal number 15 is converted to a binary number. 
+
+![Decimal to Binary Conversion](.rsrc/decimal-to-binary-example.png)
+
+Start by writing out the values for all the powers of two from the right-most digit (2^0) until you write down a number that is larger than the number you are trying to convert. From there it can be deduced that the next biggest place will be used to create the binary number. Here, we see that 16 is greater than 15, so this the final binary number will be 4 digits long. From here, you would add up 8 + 4 + 2 + 1 to get 15, so each of the 4 digits will be 1. 
+
+There are also numerous tools that convert binary numbers to decimal numbers, and vice versa. For example, [RapidTables Binary to Decimal Converter](https://www.rapidtables.com/convert/number/binary-to-decimal.html) can be used to calculate between binary and decimal numbers. 
+
+## Hexadecimal
+The hexadecimal number system uses 16 as a base, and has 16 digits represented by 0-9 and A-F. The following table shows all of the hexadecimal digits converted to decimal numbers.
+
+|Hexadecimal|Decimal|
+|----|----|
+|0|0|
+|1|1|
+|2|2|
+|3|3|
+|4|4|
+|5|5|
+|6|6|
+|7|7|
+|8|8|
+|9|9|
+|A|10|
+|B|11|
+|C|12|
+|D|13|
+|E|14|
+|F|15|
+
+There is no standard on whether the A-F digits are lowercase or uppercase, so you may see both representations. However, the **0x** prefix indicates that it is a hexadecimal number. For example, you may see 0xA4B, 0xA4b, or 0xa4b. 
+
+Hexadecimal numbers are often used because it is more human-friendly than binary numbers. Each hexadecimal digit can be used to represent four bits (or four binary digits). One common use case is using hexadecimal numbers from 0x00 to 0xFF to represent an 8-bit byte with values ranging from 0b00000000 to 0b11111111. This also saves many digits, as it only takes 2 hexadecimal digits to express decimal numbers from 0 to 255, whereas it would take 8 digits in binary. 
+
+There are also numerous tools to convert between hexadecimal numbers and decimal numbers, such as the previously mentioned tool with different setting: [RapidTables Decimal to Hexadecimal Converter](https://www.rapidtables.com/convert/number/decimal-to-hex.html). 
```

Comment:
maybe also mention cyberchef here for consistency

Reply:
Updated